pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -219,11 +229,17 @@ func (c *Reconciler) updateStatus(desired *v1alpha1.Service) (*v1alpha1.Service,
 	if reflect.DeepEqual(service.Status, desired.Status) {
 		return service, nil
 	}
+	becomesRdy := desired.Status.IsReady() && !service.Status.IsReady()
 	// Don't modify the informers copy.
 	existing := service.DeepCopy()
 	existing.Status = desired.Status
 	// TODO: for CRD there's no updatestatus, so use normal update.
-	return c.ServingClientSet.ServingV1alpha1().Services(desired.Namespace).Update(existing)
+	svc, err := c.ServingClientSet.ServingV1alpha1().Services(desired.Namespace).Update(existing)
+	if err == nil && becomesRdy {
+		duration := time.Now().Sub(svc.ObjectMeta.CreationTimestamp.Time)
+		c.statsReporter.ReportDuration(reconciler.ServiceTimeUntilReadyM, duration)
```

Comment:
Logging this information would be useful as well.
```
logger.Infof("Service %v became ready after %v"...)

Reply:
Done